eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Plutus/Evaluate.hs

```diff
@@ -382,76 +387,55 @@ evalTxExUnitsWithLogs ::
   --
   --  Unlike `evalTxExUnits`, this function also returns evaluation logs, useful for
   --  debugging.
-  Either (ContextError era) (RedeemerReportWithLogs era)
-evalTxExUnitsWithLogs pp tx utxo epochInfo sysStart = do
-  ptrToPlutusScript <-
-    forM neededPlutusScripts $ \(plutusPurpose, mPlutusScript, scriptHash) -> do
-      mPlutusScriptAndContext <-
-        forM mPlutusScript $ \plutusScript -> do
-          scriptContext <-
-            mkPlutusScriptContext
-              plutusScript
-              plutusPurpose
-              pp
-              epochInfo
-              sysStart
-              utxo
-              tx
-          pure (plutusScript, scriptContext)
-      let pointer = hoistPlutusPurpose toAsIx plutusPurpose
-      pure (pointer, (plutusPurpose, mPlutusScriptAndContext, scriptHash))
-  pure $ Map.mapWithKey (findAndCount $ Map.fromList ptrToPlutusScript) rdmrs
+  RedeemerReportWithLogs era
+evalTxExUnitsWithLogs pp tx utxo epochInfo systemStart = Map.mapWithKey findAndCount rdmrs
   where
+    usePointerAsKey (plutusPurpose, _) = hoistPlutusPurpose toAsIx plutusPurpose
+    ptrToPlutusScript = fromElems usePointerAsKey scriptsNeeded
```

Comment:
I find the pointer/ptr terminology quite confusing. What do you think about:
`keyedByPurpose` instead of `usePointerAsKey` and `purposeToScriptHash` instead  of `ptrToPlutusScript` ?
```suggestion
    keyedByPurpose (plutusPurpose, _) = hoistPlutusPurpose toAsIx plutusPurpose
    purposeToScriptHash = fromElems keyedByPurpose scriptsNeeded
```